MrE/src/main/java/frc/robot/commands/DriveCommand.java

```diff
@@ -24,142 +24,149 @@
-            }
-            if (choice == 3) { //non polynomic
-                rotation = rotation * 0.5 + Math.pow(3,(rotation * 0.5));
-            }
-            return rotation;
-        }
-
-    public void driveMode(double speed, double rotation, int choice){
-        if (choice == 0){ //semi-curvature
-            m_drivetrain.curvatureDrive(speed,rotation,true);
-        }
-        if (choice == 1){ //regular curvature
-            m_drivetrain.curvatureDrive(speed, rotation, false);
-        }
-        if (choice == 2){ //arcade
-            m_drivetrain.arcadeDrive(speed, rotation);
+        if (choice == 0){ //default left stick controls
+            speed = (m_driveController.getLeftY() * DriveConstants.kSpeedOutputModifier);
+        } else { //accelerated with trigger controls
+            speed = ((m_driveController.getLeftTriggerAxis() - m_driveController.getRightTriggerAxis()) * DriveConstants.kSpeedOutputModifier);
         }
+        return speed;
     }
 
-    
+	/**
+	 * Modifies the drive speed based on the selected user's acceleration
+	 * preference.
+	 * 
+	 * @param speed              - The current speed.
+	 * @param accelerationOption - The user selected option of acceleration.
+	 * @return The new speed based on the user's selected {@link DriveScaling}.
+	 */
+	public double scaleDriveSpeed(double speed, int accelerationOption) {
+		if (accelerationOption == 0) { // accelerate with joystick.
+			return (m_driveController.getLeftY() * DriveConstants.kSpeedOutputModifier);
+		} else { // accelerate with trigger.
+			return ((m_driveController.getLeftTriggerAxis() - m_driveController.getRightTriggerAxis())
+					* DriveConstants.kSpeedOutputModifier);
+		}
+	}
+
+	/**
+	 * Scales the speed based on the selected {@link DriveScaling}.
+	 * 
+	 * @param speed        - The current speed input to be scaled.
+	 * @param driveScaling - The {@link DriveScaling} being used.
+	 * @return The speed adjusted to the given {@link DriveScaling}.
+	 */
+	public double scaleSpeed(double speed, DriveScaling driveScaling) {
+		switch (driveScaling) {
+		case CUBIC:
+			return speed;
+		case LIMITED_POLYNOMIC:
+			double halfSpeed = speed / 2;
+			return halfSpeed + Math.pow(3, halfSpeed);
+		case QUADRATIC:
+			double speedSquared = Math.pow(speed, 2);
+			return Math.copySign(speedSquared, speed);
+		case SQUARED:
+			return Math.pow(speed, 3);
+		default:
+			throw new IllegalStateException(driveScaling + " drive scaling is not supported when scaling speed.");
+		}
+	}
+
+	/**
+	 * Scales the rotation based on the selected {@link DriveScaling}.
+	 * 
+	 * @param rotation     - The current rotation input to be scaled.
+	 * @param driveScaling - The {@link DriveScaling} being used.
+	 * @return The rotation adjusted to the given {@link DriveScaling}.
+	 */
+	public double scaleRotation(double rotation, DriveScaling driveScaling) {
+		switch (driveScaling) {
+		case CUBIC:
+			return rotation;
```

Comment:
Normally each case is ended with a break;. If a case is not ended with a break the code will also execute the next case as well. Here we are returning which acts as our break.